component/attr_cache/attr_cache.go

```diff
@@ -315,18 +292,18 @@ func (ac *AttrCache) CreateDir(options internal.CreateDirOptions) error {
 		ac.cacheLock.Lock()
 		defer ac.cacheLock.Unlock()
 		if ac.cacheDirs {
+			// the cloud is not directory-aware
 			// check if directory already exists
-			if ac.pathExistsInCache(options.Name) {
+			dirAttrCacheItem, getErr := ac.cacheMap.get(options.Name)
```

Comment:
The difference is the error being returned.
We can check if the directory already exists, but if `ac.cacheDirs` is false, we're not supposed to return an error. Because then the cloud is the authority on whether this operation succeeded.
So if `ac.cacheDirs` is false, we don't care if the directory already exists. It's being replaced.